Core/GDCore/Events/CodeGeneration/EventsCodeGenerator.cpp

```diff
@@ -468,6 +468,14 @@ gd::String EventsCodeGenerator::GenerateActionCode(
         action, *this, context);
   }
 
+  // Get the correct function name depending on whether it should be async or
+  // not
+  const gd::String& functionCallName =
+      instrInfos.IsAsync() &&
+              (!instrInfos.IsOptionallyAsync() || action.IsAwaited())

```

Comment:
This expression is used at multiple places in the code. It is not very dry, but it is annoying since it has to use both the Instruction and InstructionMetadata, which are decoupled. Would it be possible to make the gd::Instruction fetch and cache the corresponding Instruction metadata when setting the instruction type? That way, not only could this expression be implemented as a method on gd::Instruction (e.g. something like `bool gd::Instruction::RunsAsAsync()`), but from the IDE and other places it would be faster and more convenient to have the instruction metadata accessible directly instead of having to gp get it from the metadata provider every time.

Reply:
I think it's better to keep it like this:
- it's not super long
- it's super clear about what it does
- there is no hidden complexity, it's just boolean checks.

But this could be abstracted into a `ShouldRunAsAsync(instrMetadata, instruction)` static function inside a class, I'm not against it.

I would avoid at all cost caching the instruction metadata, because caching implies that:
- you should invalidate the cache when the instruction metadata is updated (which can happen for extensions).
- there is a risk that if the cache is not invalidated of having wrongly generated instructions and more generally terrible bugs.
- if the performance of something is tied to the fetching of metadata, then it's the metadata provider that should be replaced to give access in constant time to an instruction metadata (this could speed up export).

Overall, keep things "simple and stupid": 
- either keep this everywhere
- or create a quick ShouldRunAsAsync helper (which would be a good idea I guess!)
- but no caching that will have implications way beyond this (cache invalidation, yet-another-abstraction, risks of crashs...), all of this to avoid 3 checks or a function that are already super simple and clear :)